### PR TITLE
fix(environment): Clone IDF release branches instead of downloading zip and then fastforward

### PR DESCRIPTION
**Pull request by jakub-kocka**
When the selected IDF branch is the `release` one, perform `git clone` instead of downloading a `zip` archive and `fast-forwarding` to the latest commit

### Releated
* based on the Mattermost discussion (mention in tracker)
* internal tracker: IDF-10228

----
### Tested
* online IDF Installer (GitHub, Gitee and JihuLab mirrors)
    - release branch (zip archive correctly NOT downloaded -> git clone performed)
    - master branch  (zip archive correctly NOT downloaded -> git clone performed)
    - released IDF version  (zip archive correctly downloaded)
* offline IDF Installer (working as usual without any issues)

### CODE REVIEW

**Comment by jakub-kocka** [comment]
FYI @leeebo

Honestly, I’m not entirely satisfied with this approach, but given the "feature freeze" on the Windows IDF Installer, it seems like a reasonable compromise without requiring a more complex refactor.

That said, I’m open to any suggestions if you have a better approach in mind guys.

File: src/InnoSetup/Environment.iss
Comment (igrr) [inline src/InnoSetup/Environment.iss:359]: Seems like this whole code block will never be executed now. Instead of adding new code, could you simply remove the existing code which handles the fast-forwarding logic?

(And all the supporting functions, for example GitRepoDissociate and others)

**Review by igrr — CHANGES_REQUESTED** [review]
LGTM but I would prefer to remove the code rather than adding conditions and leaving the unused code there.

By the way, do you have any benchmarks, how much time did the old approach take and how much time does the new one take? And I guess it would be good to try this from China as well.

**Review by dobairoland — APPROVED** [review]
LGTM

File: src/InnoSetup/Environment.iss
Comment (igrr) [inline src/InnoSetup/Environment.iss:406]: I think you can also delete this code a few lines above:
```
    if IDFTempPath <> '' then
      CmdLine := CmdLine + ' --reference ' + IDFTempPath;
```

because IDFTempPath is never set in the `if NeedToClone` branch.

File: src/InnoSetup/PreInstall.iss
Comment (igrr) [inline src/InnoSetup/PreInstall.iss:434]: Now that you aren't calling this function for branches, a lot of code can be removed from IDFAddDownload:
https://github.com/espressif/idf-installer/blob/1ae9e300475f332fa1045708b5298e78c4bd28d3/src/InnoSetup/Environment.iss#L176-L196

File: src/InnoSetup/Environment.iss
Comment (igrr) [reply src/InnoSetup/Environment.iss]: We should still use archive download for release versions (tags), I think the change is only to do `git clone` for release branches — instead of downloading a nearest release version and then fetching the changes between it and the tip of the release branch.

**Comment by jakub-kocka** [comment]
@igrr, thanks for the review! I have implemented the changes as you suggested so it should be ready to merge. I have tested the installer and it works as expected.

Unfortunately, I am not able to do the merge as we agreed without your re-review and approval since you are the maintainer and requested a change.

**Review by leeebo — COMMENTED** [review]
@jakub-kocka LGTM! Thanks!

**Pull request merged**
